dakara_server/playlist/management/commands/createplayer.py

```
@@ -7,31 +7,108 @@
-        if not username:
-            username = "player"
 
-        # get password
-        password = getpass()
+        parser.add_argument(
+            "--username",
+            help="Specifies the loging for the player. Default to '{}'.".format(
+                USERNAME_DEFAULT
+            ),
+        )
+        parser.add_argument("--password", help="Specifies the password for the player.")
+        parser.add_argument(
+            "--noinput",
+            help="Tells Django to NOT prompt the user for input of any kind. "
+            "Use command line arguments only.",
+            action="store_true",
+        )
+
+    @staticmethod
+    def get_username():
+        """Get username from user
+
+        Returns:
+            str: Username.
+        """
+        username = input("Username (default: '{}'): ".format(USERNAME_DEFAULT))
+        return username or USERNAME_DEFAULT
+
+    def get_password(self):
+        """Get password from user
+
+        Returns:
+            str: Password.
+        """
+        while True:
+            password = getpass()
+            password_confirm = getpass("Password (again): ")
+
+            if not password == password_confirm:
+                self.stderr.write("Error: Your passwords didn't match.")
+                continue
+
+            return password
+
+    def create_player(self, username, password):
+        """Create player from provided credentials
+
+        Args:
+            username (str): Username for the player.
+            password (str): Password for the player.
+        """
+        # check password
+        if not password:
+            self.stderr.write("Error: Blank passwords aren't allowed.")
+            return
 
         try:
             player = UserModel.objects.create_user(username, password=password)
 
-        except IntegrityError:
-            self.stdout.write("Account '{}' already exists".format(username))
+        except (IntegrityError, ValueError):
```

Except an `IntegrityError` is too broad and can hide other errors.